### PR TITLE
feat(webvitals): Discover aggregate performance_score function

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds a `performance_score` function that calculates the aggregate performance score of a single web vital.
Only implemented for the metrics dataset for now (non metrics layer).

### CODE REVIEW

File: src/sentry/search/events/datasets/metrics.py [hunk 1]
```diff
@@ -1221,6 +1240,46 @@ def _resolve_web_vital_function(
...
+                        Function("equals", [Column("metric_id"), metric_id]),
+                    ],
+                ),
+                Function(
+                    "sumIf",
+                    [
+                        Column("value"),
+                        Function("equals", [Column("metric_id"), weight_metric_id]),
+                    ],
+                ),
```
Comment (k-fish) [inline src/sentry/search/events/datasets/metrics.py:1278]: does this make it `nan` if the weight is 0?

Comment (k-fish) [inline src/sentry/search/events/datasets/metrics.py:1278]: So this is `sum(individual-weighted score)  / sum (total weight)` right? That sounds correct to a ratio of individual-score to total achievable score for that metrics (it's total weight).

**Review by k-fish — COMMENTED** [review]
Generally looks good, I think this discover function is correct, I'd maybe add a couple more checks:
- Weight is 0
- Do we have any checks for score > weight?

File: src/sentry/search/events/datasets/metrics.py (hunk 1 above)
Comment (edwardgou-sentry) [reply src/sentry/search/events/datasets/metrics.py:1278]: If the numerator is > 0, then it returns none. If the numerator is 0 or there is no score, then it returns 0. Updated the test in this pr to also capture these scenarios.

Comment (edwardgou-sentry) [reply src/sentry/search/events/datasets/metrics.py:1278]: Yup, that's correct!

**Review by shruthilayaj — APPROVED** [review]

**Pull request merged**